e2e/test/iothub/properties/PropertiesWithComponentsE2ETests.cs

```diff
@@ -0,0 +1,502 @@
+                .ConfigureAwait(false);
+            Twin serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            Assert.IsTrue(serviceTwin.Properties.Reported.Contains(ComponentName));
+            Assert.IsFalse(serviceTwin.Properties.Reported[ComponentName].Contains(propName1));
+
+            await deviceClient
+                .UpdateClientPropertiesAsync(
+                    new ClientPropertyCollection
+                    {
+                        {
+                            ComponentName, propName1, new Dictionary<string, object>
+                            {
+                                [propName2] = null
+                            }
+                        }
+                    })
+                .ConfigureAwait(false);
+            serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            Assert.IsTrue(serviceTwin.Properties.Reported.Contains(ComponentName));
+            Assert.IsTrue(serviceTwin.Properties.Reported[ComponentName].Contains(propName1));
+            string value1 = serviceTwin.Properties.Reported[ComponentName][propName1].ToString();
+
+            Assert.AreEqual(value1, propEmptyValue);
+        }
+
+        private async Task PropertiesWithComponents_ClientHandlesRejectionInvalidPropertyNameAsync(Client.TransportType transport)
+        {
+            var propName1 = "$" + Guid.NewGuid().ToString();
+            var propName2 = Guid.NewGuid().ToString();
+
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(Logger, _devicePrefix).ConfigureAwait(false);
+            using var registryManager = RegistryManager.CreateFromConnectionString(Configuration.IoTHub.ConnectionString);
+            using var deviceClient = DeviceClient.CreateFromConnectionString(testDevice.ConnectionString, transport);
+
+            var exceptionThrown = false;
+            try
+            {
+                await deviceClient
+                    .UpdateClientPropertiesAsync(
+                        new ClientPropertyCollection
+                        {
+                            {
+                                ComponentName,
+                                new Dictionary<string, object> {
+                                    { propName1, 123 },
+                                    { propName2, "abcd" }
+                                }
+                            }
+                        })
+                    .ConfigureAwait(false);
+            }
+            catch (IotHubException)
+            {
+                exceptionThrown = true;
+            }
+
+            Assert.IsTrue(exceptionThrown, "IotHubException was expected for updating reported property with an invalid property name, but was not thrown.");
+
+            Twin serviceTwin = await registryManager.GetTwinAsync(testDevice.Id).ConfigureAwait(false);
+            Assert.IsFalse(serviceTwin.Properties.Reported.Contains(propName1));
```

Comment:
shouldn't this assertion be `serviceTwin.Properties.Reported[ComponentName].Contains(propName1)`?